mrmustard/physics/representations.py

```diff
@@ -155,9 +156,13 @@ def fock_array(self, shape: int | Sequence[int], batched=False) -> ComplexTensor
                     f"Expected Fock shape of length {num_vars}, got length {len(shape)}"
                 ) from e
             arrays = self.ansatz.reduce(shape).array
-        array = math.sum(arrays, axis=0)
-        arrays = math.expand_dims(array, 0) if batched else array
-        return arrays
+        arrays = math.astensor(arrays)
+        if not batched:
+            return arrays[0]
+        else:
+            array = math.sum(arrays, axis=0)
+            arrays = math.expand_dims(array, 0)
+            return arrays
```

Comment:
```suggestion
            return math.expand_dims(math.sum(arrays, axis=0), 0)
```
To make it more compact